packages/next/src/build/webpack/plugins/flight-manifest-plugin.ts

```diff
@@ -334,94 +335,75 @@ export class ClientReferenceManifestPlugin {
       // A page's entry name can have extensions. For example, these are both valid:
       // - app/foo/page
       // - app/foo/page.page
-      // Let's normalize the entry name to remove the extra extension
-      const groupName = /\/page(\.[^/]+)?$/.test(entryName)
-        ? entryName.replace(/\/page(\.[^/]+)?$/, '/page')
-        : entryName.slice(0, entryName.lastIndexOf('/'))
+      if (/\/page(\.[^/]+)?$/.test(entryName)) {
+        manifestEntryFiles.push(entryName.replace(/\/page(\.[^/]+)?$/, '/page'))
+      }
+
+      // Special case for the root not-found page.
+      if (/^app\/not-found(\.[^.]+)?$/.test(entryName)) {
+        manifestEntryFiles.push('app/not-found')
+      }
+
+      // Group the entry by their route path, so the page has all manifest items
+      // it needs:
+      // - app/foo/loading -> app/foo
+      // - app/foo/page -> app/foo
+      // - app/(group)/@named/foo/page -> app/foo
+      const groupName = entryName
```

Comment:
can we use `normalizeAppPath` util from `app-paths.ts` here?

Reply:
@huozhi That util removes `/page` and `/route` but keeps `/layout`. In this case we want to remove all last segments.